sdk-tests/src/test/java/io/dapr/it/resiliency/SdkResiliencyIT.java

```diff
@@ -13,153 +13,213 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static io.dapr.it.resiliency.SdkResiliencyIT.WIREMOCK_PORT;
+import static io.dapr.it.testcontainers.DaprContainerConstants.IMAGE_TAG;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@Testcontainers
+@WireMockTest(httpPort = WIREMOCK_PORT)
+@Tags({@Tag("testcontainers"), @Tag("resiliency")})
+public class SdkResiliencyIT {
+
+  public static final int WIREMOCK_PORT = 8888;
+  private static final Network NETWORK = Network.newNetwork();
+  private static final String STATE_STORE_NAME = "kvstore";
+  private static final int INFINITE_RETRY = -1;
+
+  @Container
+  private static final DaprContainer DAPR_CONTAINER = new DaprContainer(IMAGE_TAG)
+      .withAppName("dapr-app")
+      .withAppPort(WIREMOCK_PORT)
+      .withDaprLogLevel(DaprLogLevel.DEBUG)
+      .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("dapr-logs")))
+      .withAppHealthCheckPath("/actuator/health")
+      .withAppChannelAddress("host.testcontainers.internal")
+      .withNetworkAliases(List.of("dapr"))
+      .withNetwork(NETWORK);
+
+  @Container
+  private static final ToxiproxyContainer TOXIPROXY = new ToxiproxyContainer("ghcr.io/shopify/toxiproxy:2.5.0")
+      .withNetwork(NETWORK);
+
+  private static Proxy proxy;
+
+  private void configStub() {
+    stubFor(any(urlMatching("/actuator/health"))
+        .willReturn(aResponse().withBody("[]").withStatus(200)));
+
+    stubFor(any(urlMatching("/dapr/subscribe"))
+        .willReturn(aResponse().withBody("[]").withStatus(200)));
+
+    stubFor(get(urlMatching("/dapr/config"))
+        .willReturn(aResponse().withBody("[]").withStatus(200)));
+
+    // create a stub for simulating dapr sidecar with timeout of 1000 ms
+    var ms1000 = 1000;
+    stubFor(post(urlEqualTo("/dapr.proto.runtime.v1.Dapr/SaveState"))
+        .willReturn(aResponse().withStatus(204).withFixedDelay(ms1000)));
+
+    stubFor(any(urlMatching("/([a-z1-9]*)"))
+        .willReturn(aResponse().withBody("[]").withStatus(200)));
+
+    configureFor("localhost", WIREMOCK_PORT);
+  }
 
-  private static final int MAX_RETRIES = -1;  // Infinity
+  @BeforeAll
+  static void configure() throws IOException {
+    ToxiproxyClient toxiproxyClient = new ToxiproxyClient(TOXIPROXY.getHost(), TOXIPROXY.getControlPort());
+    proxy =
+        toxiproxyClient.createProxy("dapr", "0.0.0.0:8666", "dapr:3500");
```

Comment:
We should also consider randomizing the ports to avoid conflicts between tests.

Reply:
@cicoyle the `8666` port is the internal port of Toxiproxy container and the `3500` is the default configured by DaprContainer class. I think we can't to do it for Toxiproxy and to Dapr we need a change on container, am I right @salaboy?